pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -1038,15 +1089,9 @@ func TestReconcile(t *testing.T) {
 				t.Fatalf("Expected actions to be logged in the kubeclient, got none")
 			}
 
-			actions := clients.Kube.Actions()
-			var eventCount = 0
-			for _, action := range actions {
-				if action.GetVerb() == "create" && action.GetResource().Resource == "events" {
-					eventCount++
-				}
-			}
-			if d := cmp.Diff(tc.wantEvents, eventCount); d != "" {
-				t.Errorf("Event count does not match (-want, +got): %s. ", d)
+			err = checkEvents(fr, tc.name, tc.wantEvents)
+			if !(err == nil) {
+				t.Errorf(err.Error())
```

Comment:
another option here would be to let checkEvents actually fail the test by passing in t and using t.Helper

normally i like the clarity of handling the error, but in this case it looks like our error handling is really not adding any additional useful context, and i think the effect of using t.Helper might be the same as doing this?

Reply:
I can change it if you prefer so. Personally I prefer to see in the body of the test where t.Errorf (or t.Fatalf) is invoked, without having to look into other functions. But it's a minor thing.
Another minor aspect is that the caller of `checkEvents` still has the choice of t.Errorf and t.Fatalf depending on whether this check is blocking or not for further checks.

Reply:
i dont feel strongly enough to push back! i think t.Helper has pros and cons and that kind of thing is sort of an antipattern outside of tests

good point also re. error or fatal